lib/modifications.rb

```diff
@@ -504,5 +662,63 @@ def self.create_bundle_entry(resource)
         entry.request.url = resource.resourceType
         entry
     end
+
+    def self.get_reference_type(b, reference_string)
+      id = reference_string.split(':').last
+      b.entry.find { |e| e.resource.id == id }&.resource&.class
+    end
+
+    def self.get_resource_counts(bundle)
+      resource_counts = {}
+      bundle.entry.each do |entry|
```

Comment:
Prime spot to use `each_with_object` https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-each_with_object

Reply:
Good call. I've used `each_with_index` before, but this one I'm less familiar with.